24.md

```diff
@@ -0,0 +1,140 @@
+        "metadata": string, // Metadata json which must be presented as raw string here, this is required to pass signature verification at a later step
+        "tag": "remittanceRequest", // Type of LNURL
+        "remittanceData": {
+            "currency": string, // The currency the `LN SERVICE` supports
+            "priceHint": number,  // The current BTC sell price in the given currency
+            "feeHint": number // A hint of the fee `LN SERVIVE` will charge when receiving the Bitcoin and converting to currency. Number must be a percentage between 0 and 1.
+        }
+    }
+    ```
+    or
+
+    ```JSON
+    {"status": "ERROR", "reason": "error details..."}
+    ```
+
+    `metadata` json array must contain one `text/plain` entry, all other types of entries are optional.
+    `metadata` json array must contain either one `image/png;base64` entry or one `image/jpeg;base64` entry or neither.
+
+    The `metadata` json array is only allowed to contain arrays. The first item of an array inside the `metadata` array is always a string representing the metadata type while any item that follows can be of any JSON type. Implementors MUST NOT assume it will always be a string.
+
+    ```Typescript
+    [
+        [
+            "text/plain", // mandatory,
+            string // short description displayed when paying and in transaction log
+        ],
+        [
+            "text/long-desc", // optional
+            string // longer description of the payment, MAY contain newlines
+        ],
+        [
+            "image/png;base64",
+            string // base64 string, optional 512x512px PNG thumbnail which will represent this lnurl in a list or grid. Up to 136536 characters (100Kb of image data in base-64 encoding)
+        ],
+        [
+            "image/jpeg;base64", // optional
+            string // base64 string, 512x512px JPG thumbnail which will represent this lnurl in a list or grid. Up to 136536 characters (100Kb of image data in base-64 encoding)
+        ],
+        // future entries:
+        [
+            string,
+            any
+        ]
+    ]
+    ```
+
+    and be sent as a string:
+
+    ```JSON
+    "[[\"text/plain\", \"lorem ipsum blah blah\"]]"
+    ```
+
+3. `LN WALLET` displays a payment dialog where user can specify an exact sum in the currency supported by the wallet:
+
+	```
+	max can send = min(maxSendable, local estimation of how much sats the given amount is)
+	min can send = max(minSendable, local minimal value allowed by wallet)
+	```
+	Additionally, a payment dialog must include:
+	- Domain name extracted from `LNURL` query string.
+	- A way to view the metadata sent of `text/plain` format.
+
+    And it may include:
+    - An image element with the contents of `image/png` or `image/jpeg`.
+
+4. `LN WALLET` makes a GET request using
+
+	```
+	<callback><?|&>amount=<milliSatoshi><?|&>remittanceData=<urlencode({json object})>
```

Comment:
Here I would have expected the amount to be specified in fiat, matching the 'exact sum' entered in step 3?

The invoice `pr` returned in step 5 would be short-lived (1 minute?) and the receiving end guarantees to pay out the exact sum to the end user.

Reply:
I see what you're getting at. From our perspective, there could be two kinds of remittance invoice: (1) a quote for a closed amount of fiat or (2) an invoice that hints the amount of fiat without commitment. The first one would work as a 'price guarantee' which could be hard to achieve for general implementations of the protocol. The second one would move the price uncertainty to the client by providing a hint-without-guarantee of the price, and the wallet would be responsible for refetching the hint in short lived intervals (maybe 30 seconds?) so the user experience do not suffer.

I'm not saying one way or the other is better, but just sharing some of the things we thought while writing this. I'll update the PR to make this clearer. What do you think about it?

Reply:
@joostjager Updated the document to reflect a price guarantee.